cylc/flow/scheduler.py

```diff
@@ -1095,23 +1096,26 @@ def _configure_contact(self) -> None:
     def load_flow_file(self, is_reload=False):
         """Load, and log the workflow definition."""
         # Local workflow environment set therein.
-        self.config = WorkflowConfig(
-            self.workflow,
-            self.flow_file,
-            self.options,
-            self.template_vars,
-            is_reload=is_reload,
-            xtrigger_mgr=self.xtrigger_mgr,
-            mem_log_func=self.profiler.log_memory,
-            output_fname=os.path.join(
-                self.workflow_run_dir, 'log', 'flow-config',
-                workflow_files.WorkflowFiles.FLOW_FILE_PROCESSED
-            ),
-            run_dir=self.workflow_run_dir,
-            log_dir=self.workflow_log_dir,
-            work_dir=self.workflow_work_dir,
-            share_dir=self.workflow_share_dir,
-        )
+        try:
+            self.config = WorkflowConfig(
+                self.workflow,
+                self.flow_file,
+                self.options,
+                self.template_vars,
+                is_reload=is_reload,
+                xtrigger_mgr=self.xtrigger_mgr,
+                mem_log_func=self.profiler.log_memory,
+                output_fname=os.path.join(
+                    self.workflow_run_dir, 'log', 'flow-config',
+                    workflow_files.WorkflowFiles.FLOW_FILE_PROCESSED
+                ),
+                run_dir=self.workflow_run_dir,
+                log_dir=self.workflow_log_dir,
+                work_dir=self.workflow_work_dir,
+                share_dir=self.workflow_share_dir,
+            )
+        except IllegalValueError as exc:
+            raise UserInputError(exc)
```

Comment:
raise UserInputError(
                f"{exc.keys} in {self.flow_file} should be of type "
                f"'{exc.vtype}', but was {exc.value} of type "
                f"'{type(exc.value)}'"
            )